includes/admin/class-wcmypa-admin.php

```diff
@@ -842,41 +860,31 @@ public static function getDeliveryOptionsFromOrder(WC_Order $order, array $input
             }
         }
 
-        if (empty($meta)) {
-            $meta = null;
-        }
-
         // Create or update immutable adapter from order with a instanceof DeliveryOptionsAdapter
-        $meta = new WCMP_DeliveryOptionsFromOrderAdapter($meta, $inputData);
+        if (empty($meta) || ! empty($inputData)) {
+            $meta = new WCMP_DeliveryOptionsFromOrderAdapter($meta, $inputData);
+        }
 
         return $meta;
     }
 
     /**
-     * Output the delivery date.
+     * Output the delivery date if there is a date and the show delivery day setting is enabled.
      *
-     * @param DeliveryOptions $delivery_options
+     * @param DeliveryOptions $deliveryOptions
      *
      * @throws Exception
      */
-    private function printDeliveryDate(DeliveryOptions $delivery_options): void
+    private function printDeliveryDate(DeliveryOptions $deliveryOptions): void
     {
-        $string = $delivery_options->isPickup()
-            ? __("Pickup")
-            : __("Standard delivery",
-                "woocommerce-myparcel",
-                "woocommerce-myparcel"
-            );
-
-        // If show delivery day is enabled
         $showDeliveryDay = WCMYPA()->setting_collection->isEnabled(WCMYPA_Settings::SETTING_SHOW_DELIVERY_DAY);
 
-        if ($showDeliveryDay) {
+        if ($deliveryOptions->getDate() && $showDeliveryDay) {
             printf(
                 '<div class="delivery-date"><strong>%s</strong><br />%s, %s</div>',
                 __("MyParcel shipment:", "woocommerce-myparcel"),
-                $string,
-                wc_format_datetime(new WC_DateTime($delivery_options->getDate()), 'l d-m')
+                WCMP_Data::getDeliveryTypesHuman()[$deliveryOptions->getDeliveryType()],
+                wc_format_datetime(new WC_DateTime($deliveryOptions->getDate()), 'D d-m')
```

Comment:
Waarom verander je van `Sunday` naar `Sun`? Om ruimte te besparen?

Reply:
yes, het wordt anders vaak afgebroken in dat kleine kolommetje